app/src/main/java/com/github/se/signify/model/user/SaveUserToFireStore.kt

```diff
@@ -0,0 +1,52 @@
+package com.github.se.signify.model.user
+
+import android.util.Log
+import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.firestore.FirebaseFirestore
+import com.google.firebase.firestore.SetOptions
+
+fun saveUserToFireStore() {
+  val auth = FirebaseAuth.getInstance()
+  val db = FirebaseFirestore.getInstance()
+  val collectionPath = "users"
+  val logTag = "FireStore"
+
+  val currentUser = auth.currentUser
+
+  if (currentUser != null) {
+    val email = currentUser.email ?: "unknown"
+    val userId = email.split("@")[0]
+    val name = currentUser.displayName
+
+    // Create a User object to store in Firestore
+    val user =
+        hashMapOf(
+            "uid" to userId,
+            "email" to email,
+            "name" to name,
+            "friends" to emptyList<String>(), // Initialize the friends list
+            "friendRequests" to emptyList<String>() // Initialize the friend requests list
+            )
+
+    // Check if the user already exists in FireStore
+    val usersCollection = db.collection(collectionPath)
+    val userDocRef = usersCollection.document(userId)
+
+    userDocRef
+        .get()
+        .addOnSuccessListener { document ->
+          if (!document.exists()) {
+            // The user does not exist yet, so we add them
+            userDocRef
+                .set(user, SetOptions.merge())
```

Comment:
- Correctly handles both existing and non-existing users.
- Uses SetOptions.merge() to avoid overwriting existing fields.

Reply:
For improved reliability, you might want to use documentSnapshot.exists() in combination with documentSnapshot.data to validate data consistency before setting it.